powerauth-java-cmd-lib/src/main/java/io/getlime/security/powerauth/lib/cmd/steps/v3/EncryptStep.java

```diff
@@ -115,37 +121,50 @@ public StepContext<EncryptStepModel, EciesEncryptedResponse> prepareStepContext(
                 requestDataBytes
         );
 
-        final ClientEncryptor encryptor;
+        final EncryptorScope scope = switch (model.getScope()) {
+            case "activation":
+                yield EncryptorScope.ACTIVATION_SCOPE;
+            case "application":
+                yield EncryptorScope.APPLICATION_SCOPE;
+            default:
+                yield null;
+        };
+        if (scope == null) {
+            stepLogger.writeError("encrypt-error-scope", "Encrypt Request Failed", "Unsupported encryption scope: " + model.getScope());
+            stepLogger.writeDoneFailed("encrypt-failed");
+            return null;
+        }
+        fetchTemporaryKey(stepContext, scope);
+        final String temporaryPublicKey = (String) stepContext.getAttributes().get(TEMPORARY_PUBLIC_KEY);
+        final PublicKey encryptionPublicKey = temporaryPublicKey == null ?
+                model.getMasterPublicKey() :
+                KEY_CONVERTOR.convertBytesToPublicKey(java.util.Base64.getDecoder().decode(temporaryPublicKey));
 
         // Prepare the encryption header
         final EncryptorId encryptorId;
-        final PowerAuthEncryptionHttpHeader header;
-        switch (model.getScope()) {
-            case "application" -> {
+        ClientEncryptor encryptor = null;
```

Comment:
Just to be safe, making the variables `final` could be fine if the refactor is not too complicated.